Microsoft.Toolkit/Helpers/ObjectStorage/ISettingsStorageHelper.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Toolkit.Helpers
+{
+    /// <summary>
+    /// Service interface used to store data using key value pairs.
+    /// </summary>
+    /// <typeparam name="TKey">The type of keys to use for accessing values.</typeparam>
+    public interface ISettingsStorageHelper<TKey>
+        where TKey : notnull
+    {
+        /// <summary>
+        /// Retrieves a single item by its key.
+        /// </summary>
+        /// <typeparam name="TValue">Type of object retrieved.</typeparam>
+        /// <param name="key">Key of the object.</param>
+        /// <exception cref="KeyNotFoundException">Throws when the specified key is not found.</exception>
+        /// <returns>The TValue object</returns>
```

Comment:
Nit: all XML docs referring to type params should use the appropriate XML nodes:

```suggestion
        /// <returns>The <see typeparamref="TKey"/> object</returns>
```